src/components/atoms/Input/helpers/TextInput.tsx

```diff
@@ -111,6 +112,23 @@ export const TextInput: TextInput = ({
     else setLocalValue(value)
   }, [register, setValue, name, value])
 
+  // eslint-disable-next-line consistent-return
+  React.useEffect((): (() => void) | void => {
+    const button = document.querySelector('.show-password')
```

Comment:
Mmmh what if you have multiple `.show-password` buttons in the DOM? 😅  For example a password and validate password field 🤔  In react it's recommended to not make DOM manipulation. Here you could use an `React.useRef()` hook.

Reply:
we are already using the ref from `hook-form` which is generated from `register()`. That ref can't be manipulated.

now if we generate a new ref using `useRef()`, then we will lose the registered ref() from hook form. So i figured the current approach is the only way.